README.md

```diff
@@ -70,6 +70,108 @@ Messages observed from these Subjects are all immutable, not by convention but t
 
 Note that [heartbeat](http://jupyter-client.readthedocs.org/en/latest/messaging.html#heartbeat-for-kernels) is not included above, primarily because it's being thought of as something that may end up being deprecated.
 
+## What's in this repo?
+
+In addition to the spec doc itself (the text above) this repo contains convenience functions for enchannel implementations and consumers.  To use these functions install this package with npm
+
+    npm install enchannel
+
+The utility functions included are described below
+
+#### isChildMessage
+Checks to see if one message is child to another.  Accepts two arguments, parent and child, both of which are [Jupyter message objects](https://ipython.org/ipython-doc/3/development/messaging.html#general-message-format).  To use as a conditional:
+
+```js
+const enchannel = require('enchannel');
+if (enchannel.isChildMessage(parent, child)) {
+  console.log('is child');
+}
+```
+
+It will probably make more sense to use it as an observable filter.  In the example below, `parent` is a [Jupyter message object](https://ipython.org/ipython-doc/3/development/messaging.html#general-message-format) and `channels.iopub` is an RxJS observable:
+
+```js
+const enchannel = require('enchannel');
+const isChildMessage = enchannel.isChildMessage.bind(null, parent);
+const childMessages = channels.iopub.filter(isChildMessage);
+```
+
+#### createMessage
+Creates a [Jupyter message object](https://ipython.org/ipython-doc/3/development/messaging.html#general-message-format).  Accepts 3 arguments:
+
+ - username, string  
+ - session, string,  guid unique to the current session  
+ - msg_type: string, type of the message getting sent  
+
+The following is a full example that shows how connection may happen, how you'd setup the session and username, and then create and send a shutdown request:
+
+```js
+// The method to connect is specific to the enchannel backend implementation
+enchannelBackend.connect().then(channels => {
+
+  // Created once with the channels
+  const uuid = require('node-uuid');
+  const session = uuid.v4();
+  const username = process.env.LOGNAME || process.env.USER ||
+    process.env.LNAME || process.env.USERNAME;
+
+  // Create the shutdown request method
+  const enchannel = require('enchannel');
+  const shutdownRequest = enchannel.createMessage(username, session, 'shutdown_request');
```

Comment:
The way I'm going to be tempted to use this in the future is to bind:

```js
const createMessage = enchannel.createMessage.bind(null, username, session)
```

Which means that I :heart: the order here. At runtime I don't expect the username to change, though the session will.